qiskit/algorithms/optimizers/qnspsa.py

```diff
@@ -311,8 +311,18 @@ def get_fidelity(
 
         fid = ComputeUncompute(sampler)
 
+        num_parameters = circuit.num_parameters
+
         def fidelity(values_x, values_y):
-            result = fid.run(circuit, circuit, values_x, values_y).result()
+            values_x = np.reshape(values_x, (-1, num_parameters)).tolist()
+            batch_size_x = len(values_x)
+
+            values_y = np.reshape(values_y, (-1, num_parameters)).tolist()
+            batch_size_y = len(values_y)
```

Comment:
Do the primitives not accept Numpy array inputs for parameters?  That seems a bit unusual, but this is fine if they don't.

Reply:
That's probably my bad, we used that in the (Sampling)VQE but actually the primitives just need `Sequence[Sequence[float]]` which should work with 2d numpy arrays 🙂

Reply:
Numpy arrays aren't technically `Sequence` because they don't implement the methods `index` and `count` - it's an unfortunate implication from the `collections.abc` module defining explicit mixin methods as part of the interface, and (as far as I know), there's no built-in Python type that implies the existence of the core iterable/sequence methods, but not the ones with generic definitions.